### PR TITLE
[test-framework] Refactoring pilot test application

### PR DESCRIPTION
**Pull request by nmittler**
Simplifying the structure to make the code more readable and to make it
better fit in with the application model of the new local agent.

### CODE REVIEW

File: pilot/test/server/echo/echo.go [hunk 1]
```diff
@@ -1,350 +1,199 @@
...
+	return nil
 }
 
-// Imagine a pie of different flavors.
-// The flavors are the HTTP response codes.
-// The chance of a particular flavor is ( slices / sum of slices ).
-type codeAndSlices struct {
-	httpResponseCode int
-	slices           int
+type serverInterface interface {
```
Comment (ozevren) [inline pilot/test/server/echo/echo.go]: Nit: Please add a descriptive comment to the interface.

File: pilot/test/server/echo/echo.go [hunk 2]
```diff
@@ -1,350 +1,199 @@
...
+	server *grpc.Server
 }
 
-func setResponseFromCodes(request *http.Request, response http.ResponseWriter) error {
-	responseCodes := request.FormValue("codes")
-
-	codes, err := validateCodes(responseCodes)
+func (s *grpcServer) start() error {
+	// Listen on the given port and update the port if it changed from what was passed in.
+	listener, p, err := listenOnPort(s.port.Port)
```
Comment (ozevren) [inline pilot/test/server/echo/echo.go]: Is this to handle s.port.Port = 0 case?

File: pilot/test/server/echo/handler.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		// allow all connections by default
+		return true
+	},
+} //defaults
+
+// Imagine a pie of different flavors.
```
Comment (ozevren) [inline pilot/test/server/echo/handler.go]: interesting analogy

Comment (nmittler) [reply pilot/test/server/echo/handler.go]: Haha .... not my comment ... I've never seen this before.  But hilarious! :)

File: pilot/test/server/echo/echo.go (hunk 2 above)
Comment (nmittler) [reply pilot/test/server/echo/echo.go]: yes, exactly

**Pull request merged**